### PR TITLE
refactor: update classnames usage to not use object style

### PR DESCRIPTION
**Pull request by HeartSquared**
## Why

Consumers using NextJS have told us that they are getting TypeScript errors relating to our styles.
eg. https://github.com/cultureamp/unified-navigation/actions/runs/4905384570/jobs/8759130223?pr=777
image

I have deduced it is likely related to the object style used within `classnames` expecting the key (classname) to be a string, but not finding the type, and therefore rejecting the `unknown` type (we define the type in our root `types.d.ts`, which isn't exported).

## What

- Update `classnames` usage to not use object style
- Rename any aliases `classNames` to `classnames`
- Boyscout where it was easy

### CODE REVIEW

File: docs/Systems/Tailwind/UtilityClassReferences/Borders/border-spacing.stories.tsx [hunk 1]
```diff
@@ -37,20 +36,12 @@ export const BorderSpacing: StoryFn<{ isReversed: boolean }> = ({
       <table className={utilityClass}>
         <tbody>
           <tr>
-            <td className={classnames("outline-[black] outline-1 outline")}>
-              Tutant
-            </td>
-            <td className={classnames("outline-[black] outline-1 outline")}>
-              Meenage
-            </td>
+            <td className={"outline-[black] outline-1 outline"}>Tutant</td>
```
Comment (gyfchong) [inline docs/Systems/Tailwind/UtilityClassReferences/Borders/border-spacing.stories.tsx:39]: This doesn't even need curlies

Comment (HeartSquared) [reply docs/Systems/Tailwind/UtilityClassReferences/Borders/border-spacing.stories.tsx:39]: https://github.com/cultureamp/kaizen-design-system/pull/3609

BAM!

File: draft-packages/title-block-zen/KaizenDraft/TitleBlockZen/TitleBlockMenuItem.tsx [hunk 2]
```diff
@@ -14,11 +14,12 @@ export type TitleBlockMenuItemProps =
...
-  const className = classnames(styles.menuItem, {
-    [styles["menuItem--destructive"]]: styles.destructive,
-    [styles["menuItem--disabled"]]: styles.disabled,
-    [styles["menuItem--active"]]: styles.isActive,
-  })
+  const className = classnames(
+    styles.menuItem,
+    styles.destructive && styles["menuItem--destructive"],
+    styles.disabled && styles["menuItem--disabled"],
+    styles.isActive && styles["menuItem--active"]
```
Comment (HeartSquared) [inline draft-packages/title-block-zen/KaizenDraft/TitleBlockZen/TitleBlockMenuItem.tsx:21]: Wait what? What is happening here? 😅

Comment (gyfchong) [reply draft-packages/title-block-zen/KaizenDraft/TitleBlockZen/TitleBlockMenuItem.tsx:21]: @HeartSquared hurt herself in her confusion

**Comment by gyfchong** [comment]
✌🏼

**Review by gyfchong — APPROVED** [review]

File: packages/typography/src/Paragraph/Paragraph.tsx [hunk 3]
```diff
@@ -53,18 +53,16 @@ export const Paragraph = ({
...
+  )
 
   return createElement(
     tag === undefined ? "p" : tag,
     { ...restProps, className },
     children
   )
 }
-
-Paragraph.displayName = "Paragraph"
```
Comment (dougmacknz) [inline packages/typography/src/Paragraph/Paragraph.tsx:70]: Did you mean to remove `displayName` from some components in this PR @HeartSquared?

**Pull request merged**